tsconfig.json

```diff
@@ -18,6 +18,7 @@
     "noImplicitAny": false,
     "types": ["node", "requestidlecallback"],
     "baseUrl": "./",
+    "skipLibCheck": true,
```

Comment:
I had a look locally for why this line. I get:

<img width="1354" alt="Capture d’écran 2021-07-14 à 21 50 04" src="https://user-images.githubusercontent.com/3165635/125684010-88638bff-f656-4774-be30-0adb78163eb9.png">

Reply:
We had a similar problem in #2106. The docs in the core already enables this flag: https://github.com/mui-org/material-ui/blob/8a4b8952077b64e90ac9d23c116a45aed672db9c/docs/tsconfig.json#L11 TypeScript also encourages to set it to true: https://www.typescriptlang.org/tsconfig#skipLibCheck

If we don't want to encourage enabling this flag, we should address this in the core.

Reply:
> We had a similar problem in #2106

I assume it's a reference to `Property 'components' does not exist on type 'Localization'.`. It seems to be the problem in reverse (thrid-party that tries to use data-grid but fail because its internal types are wrong). I have faced it when working on https://github.com/mui-org/material-ui/pull/27196, and fixed it in #2117.

Reply:
Regarding `skipLibCheck`, TypeScript has the value to false by default. My concern is with developers that would complain because we break their build.

Reply:
We don't need the `skipLibCheck` anymore, I already removed it. Since we created our local classes utility there're no more errors from @material-ui/unstyled.